lib/JitEngineDeviceCUDA.cpp

```diff
@@ -120,10 +121,26 @@ void JitEngineDeviceCUDA::setLaunchBoundsForKernel(Module &M, Function &F,
   // properties.
   // TODO: set min GridSize.
   int MaxThreads = std::min(1024, BlockSize);
-  Metadata *MDVals[] = {ConstantAsMetadata::get(&F),
-                        MDString::get(M.getContext(), "maxntidx"),
-                        ConstantAsMetadata::get(ConstantInt::get(
-                            Type::getInt32Ty(M.getContext()), MaxThreads))};
+  auto *FuncMetadata = ConstantAsMetadata::get(&F);
+  auto *MaxntidxMetadata = MDString::get(M.getContext(), "maxntidx");
+  auto *MaxThreadsMetadata = ConstantAsMetadata::get(
+      ConstantInt::get(Type::getInt32Ty(M.getContext()), MaxThreads));
+
+  // Replace if the metadata exists.
+  for (auto *MetadataNode : NvvmAnnotations->operands()) {
+    // Expecting 3 operands ptr, desc, i32 value.
+    assert(MetadataNode->getNumOperands() == 3);
+
+    auto *PtrMetadata = MetadataNode->getOperand(0).get();
+    auto *DescMetadata = MetadataNode->getOperand(1).get();
+    if (PtrMetadata == FuncMetadata && MaxntidxMetadata == DescMetadata) {
+      MetadataNode->replaceOperandWith(2, MaxThreadsMetadata);
+      return;
+    }
+  }
```

Comment:
Is this what we want? Are we going to override the developers intend? This sounds as bad practice. @davidbeckingsale what do you think?

Reply:
Yes, we are overriding with a value that is correct to achieve greater optimization

Reply:
You don't know that though. What if the user has auto-tuned the code and knows that these values optimize the code? There is no guarantee on what you do there.

Reply:
It's possible to turn off this optimization with an env var. Down the road we can add annotation extensions to turn it on/off per kernel

Reply:
IMO this is not intuitive. The developer says:
```
__global__ __attribute__((annotate("jit")))
__launch_bounds__(128, 4)
```
this directly translates to  explicit control to turn it on/off per kernel with no additional support from us.

Reply:
User can be mistaken, Proteus isn't 😝